src/bootstrap/lib.rs

```diff
@@ -867,6 +871,12 @@ impl Build {
         return base
     }
 
+    /// Returns if this target should statically link the C runtime, if specified
+    fn crt_static(&self, target: &str) -> Option<bool> {
+        self.config.target_config.get(target)
+            .and_then(|t| t.crt_static)
+    }
```

Comment:
As an idea actually, I think we can leverage this for MSVC as well. The `bin/rustc.rs` script above has a hard-coded branch for MSVC but I think we could change that here. Could this code be updated with something like:

```rust
.or_else(|| {
    if target.contains("msvc") {
        Some(true)
    } else {
        None
    }
})